pkg/kubelet/config/http.go

```diff
@@ -183,6 +229,63 @@ func tryDecodeList(data []byte) (parsed bool, manifests []v1beta1.ContainerManif
 	return true, manifests, pods, nil
 }
 
+func tryDecodeSinglePod(data []byte) (parsed bool, pod api.Pod, err error) {
+	// TODO(wojtek-t): Should we support pods in different api versions?
+	// If so, can we do it smarter than trying to parse for every single version one by one?
+	var obj v1beta3.Pod
+	if err = yaml.Unmarshal(data, &obj); err != nil {
+		return false, pod, err
+	}
+	if err = api.Scheme.Convert(&obj, &pod); err != nil {
+		return true, pod, err
+	}
+	// TODO(wojtek-t): applyDefaults is filling few necessary fields (e.g. Namespace),
```

Comment:
We should probably do both, apply defaults and validate. What is the problem you're running into?